store/job_store.go

```diff
@@ -20,9 +24,37 @@ type DataStore struct {
 //JobsMap is a map used for storing Job resources with the keys being string values.
 var JobsMap = make(map[string]models.Job)
 
-// CreateJob creates a new Job resource and stores it in the JobsMap.
-func (ds *DataStore) CreateJob(ctx context.Context, id string) (models.Job, error) {
+//LastUpdatedSlice is a type that implements the sort interface so that the jobs in it can be sorted using the generic Sort function.
+type LastUpdatedSlice []models.Job
+
+//Len is a function that's required by the sort interface.
+func (s LastUpdatedSlice) Len() int {
+	return len(s)
+}
+
+//Less is a function that's required by the sort interface.
+func (s LastUpdatedSlice) Less(i, j int) bool {
+	return s[i].LastUpdated.Before(s[j].LastUpdated)
+}
+
+//Swap is a function that's required by the sort interface.
+func (s LastUpdatedSlice) Swap(i, j int) {
+	s[i], s[j] = s[j], s[i]
+}
+
+//DeleteAllJobs empties the JobStore by deleting everything from the JobsMap.
+func (ds *DataStore) DeleteAllJobs(ctx context.Context, mux *sync.Mutex) error {
+	log.Event(ctx, "deleting all jobs from the job store")
+	mux.Lock()
+	for k := range JobsMap {
+		delete(JobsMap, k)
+	}
+	mux.Unlock()
+	return nil
+}
 
+// CreateJob creates a new Job resource and stores it in the JobsMap.
+func (ds *DataStore) CreateJob(ctx context.Context, id string, mux *sync.Mutex) (models.Job, error) {
```

Comment:
I agree it would be worth it except that my next ticket will require me to remove all the Mutex code as I will be replacing the map with calls to Mongo dB. So I would prefer to leave things as they are. Is that ok?

Reply:
Ah ok sure, no worries